docs/ledger/migrations/v4.0-migration.md

```diff
@@ -0,0 +1,325 @@
+
+  - name change
+  - updated `info` to `batch`
+  - updated `BatchInfo` (see types)
+
+- [v1alpha1.QueryClassesRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryClassesRequest) → [v1.QueryClassesRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QueryClassesRequest)
+
+  - no changes
+
+- [v1alpha1.QueryClassesResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryClassesResponse) → [v1.QueryClassesResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QueryClassesResponse)
+
+  - updated `ClassInfo` (see types)
+
+- [v1alpha1.QueryClassInfoRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryClassInfoRequest) → [v1.Query.Class](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Query.Class)
+
+  - name change
+  - no changes to fields
+
+- [v1alpha1.QueryClassInfoRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryClassInfoRequest) → [v1.Query.Class](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.Query.Class)
+
+  - name change
+  - updated `ClassInfo` (see types)
+
+- [v1alpha1.QueryCreditTypesRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryCreditTypesRequest) → [v1.QueryCreditTypesRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QueryCreditTypesRequest)
+
+  - no changes
+
+- [v1alpha1.QueryCreditTypesResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QueryCreditTypesResponse) → [v1.QueryCreditTypesResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QueryCreditTypesResponse)
+
+  - no changes
+
+- [v1alpha1.QuerySupplyRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QuerySupplyRequest) → [v1.QuerySupplyRequest](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QuerySupplyRequest)
+
+  - no changes
+
+- [v1alpha1.QuerySupplyResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1alpha1#regen.ecocredit.v1alpha1.QuerySupplyResponse) → [v1.QuerySupplyResponse](https://buf.build/regen/regen-ledger/docs/main:regen.ecocredit.v1#regen.ecocredit.v1.QuerySupplyResponse)
+
+  - added `cancelled_amount`
+  - updated `tradable_supply` to `tradable_amount`
+  - updated `retired_supply` to `retired_amount`
+
```

Comment:
I think we should also add params query migration.

Reply:
https://github.com/regen-network/regen-ledger/pull/1189/commits/f7fdfad6b069ad6ccb053de9b433ae431795ac03